script.js

```diff
@@ -0,0 +1,618 @@
+        },
+        
+        {
+            id: '17',
+            description: 'desctiption',
+            createdAt: new Date('2021-03-15T23:00:00'),
+            link: 'link',
+            vendor: 'vendor',
+            hashTags: ['hashtag'],
+            discount: '10%',
+            validUntil: new Date('2021-06-30T23:00:00'),
+        },
+        
+        {
+            id: '18',
+            description: 'desctiption',
+            createdAt: new Date('2021-02-15T23:00:00'),
+            link: 'link',
+            vendor: 'vendor',
+            hashTags: ['hashtag'],
+            discount: '10%',
+            validUntil: new Date('2021-05-29T23:00:00'),
+        },
+        
+        {
+            id: '19',
+            description: 'desctiption',
+            createdAt: new Date('2021-01-15T23:00:00'),
+            link: 'link',
+            vendor: 'vendor',
+            hashTags: ['hashtag'],
+            discount: '10%',
+            validUntil: new Date('2021-04-29T23:00:00'),
+        },
+        
+        {
+            id: '20',
+            description: 'desctiption',
+            createdAt: new Date('2021-02-02T23:00:00'),
+            link: 'link',
+            vendor: 'vendor',
+            hashTags: ['hashtag'],
+            discount: '10%',
+            validUntil: new Date('2021-03-29T23:00:00'),
+        }
+
+    ];
+
+    function getAds(skip = 0, top = 10, filterConfig = undefined) {
+        if (typeof skip !== 'number' || typeof top !== 'number') {
+            console.log('Incorrect inputting type');
+            return;
+        }
+
+        if (filterConfig) {
+            let returningAds;
+            for (var param in filterConfig) {
+                if (param === 'hashTags') {
+                    for (var i = 0; i < filterConfig.hashTags.length; i++) {
+                        returningAds = adList.filter(adItem => adItem.hashTags.includes(filterConfig.hashTags[i]));
+                    }
+                } else if (param === 'dateFrom') {
+                    returningAds = adList.filter(adItem => adItem.createdAt >= filterConfig.dateFrom);
+                } else if (param === 'dateTo') {
+                    returningAds = adList.filter(adItem => adItem.createdAt < filterConfig.dateTo);
+                } else if (param === 'vendor') {
+                    returningAds = adList.filter(adItem => adItem.vendor === filterConfig.vendor);
+                }
+            }
+
+            returningAds.sort(function (first, second) {
+                if (first.createdAt < second.createdAt) {
+                    return 1;
+                } else if (first.createdAt > second.createdAt) {
+                    return -1;
+                } else {
+                    return 0;
+                }
+            });
+
+            return returningAds.slice(skip, skip + top);
+        } else {
+            var returningAds = adList.slice(skip, skip + top);
+
+            returningAds.sort(function (first, second) {
```

Comment:
If you see that some code block is repeated two or more times, it's always a good idea to move it to separate function 😉